File: src/SunAdmin.Domain/Entities/Role.cs
using SunAdmin.Domain.Common;
using SunAdmin.Domain.Enums;

namespace SunAdmin.Domain.Entities;

public sealed class Role : AuditableEntity
{
    public string Code { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public RoleDataScope DataScope { get; set; } = RoleDataScope.All;
    public RecordStatus Status { get; set; } = RecordStatus.Enabled;
    public bool IsBuiltIn { get; set; }
}

File: web/sun-admin-web/src/types/role.ts
import type { AuditableEntity, EntityStatus } from './api';

export interface RoleItem extends AuditableEntity {
  name: string;
  code: string;
  description?: string;
  status: EntityStatus;
  isBuiltIn?: boolean;
  menuIds?: number[];
}

export interface RoleUpsertRequest {
  name: string;
  code: string;
  description?: string;
  status: EntityStatus;
}

export interface UpdateRoleMenusRequest {
  menuIds: number[];
}
